From a413bf673d68aaf69885a4c2869a038124f63001 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Thu, 23 Jan 2014 16:40:21 +0800
Subject: [PATCH 2/4] ENGR00296049 - record 1080P video, sometimes recorded
 file's only 15 fps

Video stream and preview stream use same camera capture thread,
make sure preview frame rate is 30 fps

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 src/com/android/camera/VideoModule.java | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 0ee62bb..61cffe3 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -14,6 +14,11 @@
  * limitations under the License.
  */
 
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ */
+
+
 package com.android.camera;
 
 import android.annotation.TargetApi;
@@ -1463,14 +1468,17 @@ public class VideoModule implements CameraModule,
     @SuppressWarnings("deprecation")
     private void setCameraParameters() {
         mParameters.setPreviewSize(mDesiredPreviewWidth, mDesiredPreviewHeight);
-        int[] fpsRange = CameraUtil.getMaxPreviewFpsRange(mParameters);
+       /* int[] fpsRange = CameraUtil.getMaxPreviewFpsRange(mParameters);
         if (fpsRange.length > 0) {
             mParameters.setPreviewFpsRange(
                     fpsRange[Parameters.PREVIEW_FPS_MIN_INDEX],
                     fpsRange[Parameters.PREVIEW_FPS_MAX_INDEX]);
         } else {
             mParameters.setPreviewFrameRate(mProfile.videoFrameRate);
-        }
+        } */
+
+	    //make sure record frame rate is 30 fps
+		mParameters.setPreviewFrameRate(mProfile.videoFrameRate);
 
         forceFlashOffIfSupported(!mUI.isVisible());
 
-- 
1.8.0

